### ISSUE TITLE
TypeError: expected str, bytes or os.PathLike object, not coo_matrix

### ISSUE THREAD

**Issue by farshadf** [ratschlab/spladder#178]
* spladder version: 3.0.3
* Python version: 3.10
* Operating System: CentOS 7
### Description

Hi, 
I am trying to augment the annotation using "build" on STAR generated BAM files on the mouse reference genome, but spladder does not progress as expected although it seems to be starting fine.

### What I Did
Script:
```
annotfile='/references/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf'
spladder build --bams alignment_list.txt \
               --annotation $annotfile \
               --outdir spladder_out \
               --ignore-mismatches \
               --parallel 20 \
               --sparse-bam \
               --set-mm-tag nM
```

This is the error message:
```
No information found for contig chr4_GL456216_random
No information found for contig chr1_GL456221_random
No information found for contig chr1_GL456211_random
No information found for contig chr4_JH584292_random
No information found for contig chrUn_JH584304
No information found for contig chrX_GL456233_random
Traceback (most recent call last):
  File "HOME/base3/bin/spladder", line 8, in <module>
    sys.exit(main())
  File "HOME/base3/lib/python3.10/site-packages/spladder/spladder.py", line 229, in main
    options.func(options)
  File "HOME/base3/lib/python3.10/site-packages/spladder/spladder_build.py", line 73, in spladder
    prep_sparse_bam_filtered(options)
  File "HOME/base3/lib/python3.10/site-packages/spladder/spladder_prep.py", line 57, in prep_sparse_bam_filtered
    P_IN = h5py.File(tmp[1], 'r')
  File "HOME/base3/lib/python3.10/site-packages/h5py/_hl/files.py", line 509, in __init__
    name = filename_encode(name)
  File "HOME/base3/lib/python3.10/site-packages/h5py/_hl/compat.py", line 19, in filename_encode
    filename = fspath(filename)
TypeError: expected str, bytes or os.PathLike object, not coo_matrix
```

This alignment_list.txt:
```
/test/STAR/2501_S1_R1_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2501_S1_R2_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2502_S2_R1_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2502_S2_R2_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2503_S3_R1_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2503_S3_R2_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2504_S4_R1_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2504_S4_R2_001.fastq-trimmedAligned.sortedByCoord.out.bam
/test/STAR/2507_S5_R1_001.fastq-trimmedAligned.sortedByCoord.out.bam

```

Any help is appreciated! Thank you.

**Comment by MB571** [comment]
I'm getting the same/similar error

**Comment by akahles** [comment]
Dear @MB571 & @farshadf ,

Thanks a lot for reporting this and sorry for the late reply. Unfortunately, I sometimes fall behind on handling issues.

I have looked at the respective code part and I think I found the issue. In case of `--sparse-bam` and use of `--parallel` the encounter of unknown contig names was not cleanly handled. This should be better now. I have pushed a fix to the development branch and will make this available with the next release. 

Best,
Andre
